client/src/components/Popup.tsx

```diff
@@ -0,0 +1,32 @@
+import { ReactNode } from 'react';
+import { createPortal } from 'react-dom';
+
+type Props = {
+  children: ReactNode;
+  isOpen: boolean;
+  buttonRef?: React.RefObject<HTMLButtonElement> | null;
```

Comment:
I would prefer a name that indicates why you need this reference. And it doesn't have to be from a button - in theory, you could click on pretty much anything to bring up the popup.